c-basics/src/isOlder.cpp

```diff
@@ -44,52 +44,52 @@ bool validatedate(char *dob1, char *dob2, int *d1, int *d2)
 
 
 }
-bool validateyear(char *dob1, char *dob2, int *y1, int *y2,int *d1,int *d2,int *m1,int *m2)
+bool validateyear(char *dob1, char *dob2, int *y1, int *y2, int *d1, int *d2, int *m1, int *m2)
 {
-						if (((dob1 + 6 && *(dob1 + 6) != '-') && (dob1 + 7 && *(dob1 + 7) != '-')) && ((dob2 + 6 && *(dob2 + 6) != '-') && (dob2 + 7 && *(dob2 + 7) != '-')) && ((dob1 + 8 && *(dob1 + 8) != '-') && (dob1 + 9 && *(dob1 + 9) != '-')) && ((dob2 + 8 && *(dob2 + 8) != '-') && (dob2 + 9 && *(dob2 + 9) != '-')))
-						{
-							*d1 = ((*dob1 - 48)*10)+(*(dob1+1)-48);
-							*d2 = ((*dob2 - 48) * 10) + (*(dob2 + 1) - 48);
+	if (((dob1 + 6 && *(dob1 + 6) != '-') && (dob1 + 7 && *(dob1 + 7) != '-')) && ((dob2 + 6 && *(dob2 + 6) != '-') && (dob2 + 7 && *(dob2 + 7) != '-')) && ((dob1 + 8 && *(dob1 + 8) != '-') && (dob1 + 9 && *(dob1 + 9) != '-')) && ((dob2 + 8 && *(dob2 + 8) != '-') && (dob2 + 9 && *(dob2 + 9) != '-')))
+	{
+		*d1 = ((*dob1 - 48) * 10) + (*(dob1 + 1) - 48);
+		*d2 = ((*dob2 - 48) * 10) + (*(dob2 + 1) - 48);
 
-							*m1 = (((*(dob1 + 3) - 48) * 10) + (*(dob1 + 4) - 48));
-							*m2 = (((*(dob2 + 3) - 48) * 10) + (*(dob2 + 4) - 48));
-							*y1 = (((((((*(dob1 + 6) - 48) * 10) + (*(dob1 + 7) - 48)) * 10) + (*(dob1 + 8) - 48)) * 10) + *(dob1 + 9) - 48);
+		*m1 = (((*(dob1 + 3) - 48) * 10) + (*(dob1 + 4) - 48));
+		*m2 = (((*(dob2 + 3) - 48) * 10) + (*(dob2 + 4) - 48));
+		*y1 = (((((((*(dob1 + 6) - 48) * 10) + (*(dob1 + 7) - 48)) * 10) + (*(dob1 + 8) - 48)) * 10) + *(dob1 + 9) - 48);
 
-							*y2 = (((((((*(dob2 + 6) - 48) * 10) + (*(dob2 + 7) - 48)) * 10) + (*(dob2 + 8) - 48)) * 10) + *(dob2 + 9) - 48);
-							if ((*d1 > 31 || *d2 > 31) || (*m1 > 12 || *m2 > 12))
-								return false;
-							if (*m1 == 2)
-								{
-								if (*y1 % 400==0 || (*y1 % 100 != 0 && *y1 % 4 == 0))
-								{
-									//printf("i");
-									if (*d1 > 29)
-										return false;
-								}
-								else if ((*y1 % 4 != 0) && *d1>28)
-									return false;
-							}
-							if (*m2 == 2)
-							{
-								//printf("i");
-								if (*y2 % 400==0 || (*y2 % 100 != 0 && *y2 % 4 == 0))
-								{
-									//printf("l");
-									if (*d2 > 29)
-										return false;
-								}
-								else if ((*y2 % 4 != 0) && *d2>28)
-									return false;
-							}
-								return true;
-						}
-						else return false;
+		*y2 = (((((((*(dob2 + 6) - 48) * 10) + (*(dob2 + 7) - 48)) * 10) + (*(dob2 + 8) - 48)) * 10) + *(dob2 + 9) - 48);
+		if ((*d1 > 31 || *d2 > 31) || (*m1 > 12 || *m2 > 12))
+			return false;
+		if (*m1 == 2)
```

Comment:
I have asked you to make this modular. Can you please commit again after making those changes. No need of creating new pull requests all commits will be automatically tracked.

Reply:
I commit with requested changes. Closes #3